llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -891,13 +899,9 @@ void VPlan::execute(VPTransformState *State) {
     }
   }
 
-  // We do not attempt to preserve DT for outer loop vectorization currently.
-  if (!EnableVPlanNativePath) {
-    BasicBlock *VectorHeaderBB = State->CFG.VPBB2IRBB[Header];
-    State->DT->addNewBlock(VectorHeaderBB, VectorPreHeader);
-    updateDominatorTree(State->DT, VectorHeaderBB, VectorLatchBB,
-                        State->CFG.ExitBB);
-  }
+  State->CFG.DTU.flush();
+  assert(EnableVPlanNativePath || State->CFG.DTU.getDomTree().verify(
+                                      DominatorTree::VerificationLevel::Fast));
```

Comment:
```suggestion
  // DT is currently updated for non-native path only.
  assert(EnableVPlanNativePath || State->CFG.DTU.getDomTree().verify(
                                      DominatorTree::VerificationLevel::Fast));
```
(retaining the comment)

Reply:
Added back, thanks!